compat/os/mingw.h

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_OS_WINDOWS_MINGW_H
+#define SQUID_OS_WINDOWS_MINGW_H
+
+#if _SQUID_MINGW_
+
+/****************************************************************************
+ *--------------------------------------------------------------------------*
+ * DO *NOT* MAKE ANY CHANGES below here unless you know what you're doing...*
+ *--------------------------------------------------------------------------*
+ ****************************************************************************/
+
```

Comment:
This is compliance with the rest of the portability files. Changing policy here is out of scope for this PR.

FWIW, the statement is clear. **everything** below that line is OS-specific and cannot be safely changed without expert knowledge of the POSIX API, the mapped system API, and active testing on the relevant OS build environment.

Reply:
> > This is compliance with the rest of the portability files. Changing policy here is out of scope for this PR.
> 
> There is no such policy and no policy change is requested. 

This request is **creating** a new policy prohibiting this documentation.
Such policy change is out of scope for this PR.


> Not all portability files have this blob, so not adding it to this new file is not even going to create a new inconsistency.

All `compat/os/*.h` files should contain this warning. I see that Solaris and OpenSolaris broke the pattern. That will need fixing, but not here. This PR is maintaining consistency

> 
> > FWIW, the statement is clear. **everything** below that line is OS-specific and cannot be safely changed without expert knowledge of the POSIX API, the mapped system API, and active testing on the relevant OS build environment.
> 
> I am not against adding a statement similar to the above paragraph _if_ we want to block untested changes from non-experts. However, I doubt we want that
>. AFAICT, none of us would pass a meaningful bar in this particular context!

You are wrong. We do have the necessary expertise in the Squid Project.

> Let's document actual dependencies and other critical caveats (where it makes sense) rather than making grandiose statements about general dangers of modifying low-level OS-specific code.

No need for insults ("grandious"). This is not a general danger - this is very specific danger for these `compat/os/*.h` which need a very specific environment and toolchain and testing process to get any changes right. The same cannot be said of other Squid code.


> A lot of Squid code requires expert knowledge for serious modifications. Sprinkling code with such notices is likely to scare valuable contributions before we can review them.

In libcompat the definition of "valuable contribution" has somewhat stricter criteria than even the most complex other part of the Squid code. Making people reluctant to touch this particular code is the purpose.

Reply:
Lets leave `os/mingw.h` matching the other operating system `os/*.h` files and save some brain cells.

Reply:
Indeed you are cycling. I have already addressed that seeming inconsistency.